pkg/storage/unified/search/document.go

```diff
@@ -0,0 +1,166 @@
+package search
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/services/store/kind/dashboard"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+)
+
+type DocumentBuilderProvider interface {
+	GetDocumentBuilders(ctx context.Context) ([]resource.DocumentBuilderInfo, error)
```

Comment:
^^^ not sure if this should be in this package or in `resource` -- long term, the known resource types will likely have an "app-manifest" that needs to be loaded to know what can be parsed